awswrangler/_databases.py

```diff
@@ -60,11 +60,18 @@ def _get_connection_attributes_from_catalog(
                     f"No CA certificate found at {ssl_cert_path}."
                 )
         ssl_context = ssl.create_default_context(cadata=ssl_cadata)
-
+    if details.get("USERNAME") is None:
+        user = details["username"]
```

Comment:
It looks like according to the [docs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/glue.html#Glue.Client.get_connection), if the credentials are stored in secrets manager, there will be `SECRET_ID` key and the credentials would have to be retrieved from Secrets Manager separately using `wr.secretscmanager.get_secret` or boto3.